Java: Add FAQ Entry for Spring WebFlux (#362)
SAP/cloud-sdk: docs/java/faq.mdx

We still provide full support for SAP Business Technology Platform Neo.
However, we do not recommend starting new projects with SAP BTP for the Neo environment, as the Cloud Foundry environment is better suited for apps and SAP S/4HANA extension development.
Also, for Neo we provide out-of-the-box support with only servlet based applications and not spring based applications.

### Do You Support Reactive Spring Technology, e.g. Webflux?

> The SAP Cloud SDK for Java currently does not fully work with the [Reactive Spring Technology Stack](https://docs.spring.io/spring-framework/docs/5.0.0.M1/spring-framework-reference/html/web-reactive.html).

Reviewer: ```suggestion
The SAP Cloud SDK for Java currently is not fully compatible with the [Reactive Spring Technology Stack](https://docs.spring.io/spring-framework/docs/5.0.0.M1/spring-framework-reference/html/web-reactive.html).
```

> Most importantly the [Multi Tenancy Features](features/multi-tenancy/multi-tenancy-thread-context) are not functional on that stack.
> The SAP Cloud SDK for Java and its multi tenancy features are build on a servlet based concept which is incompatible with the reactive stack.

Reviewer 2: What do you think about that:

```suggestion
Most importantly, the [Multi Tenancy Features](features/multi-tenancy/multi-tenancy-thread-context) are not functional on that stack, because they were designed with servlet based applications in mind.
```

Reads just a little bit more fluent IMO.

> The SAP Cloud SDK for Java and its multi tenancy features are build on a servlet based concept which is incompatible with the reactive stack.

Reviewer: Is it smth like the classical threads-based concurrency model? Servlets are usually mentioned more in the context of Web apps?

Reviewer 3: yes exactly
